### ISSUE TITLE
reverse proxies such as nginx

### ISSUE THREAD

**Issue by airdogvan** [rootless-containers/rootlesskit#176]
I was wondering, any special advice for using the above mentioned?
When I try in rootless userspace the proxies seem to generate all the right entries, no errors in the logs but the containers do not get routed, with the result that the destination domains cannot be reached.
So I'm guessing this has something to do with how rootless handles networking.
Read something about disabling userland proxy but although haven't tried it yet doesn't look to me that it'll solve the issue for me.
I've also read about the different network drivers mentioned in the docs here, haven't tried them (now getting real late here) and not sure would solve my problem.
Any hints or better magical solutions welcomed.

It's all the more frustrating as on machines (behind physical routers) where I don't need proxies all containers run very well but where proxies are needed they form the basis of the pyramid and if that doesn't work not really worth it to bother about the rest.

**Comment by AkihiroSuda** [comment]
Is this about rootless Docker? Podman? Please post a full reproducer of the issue

**Comment by airdogvan** [comment]
I installed rootless docker according to these docs: https://docs.docker.com/engine/security/rootless/.
Running Docker version 19.03.12 on Ubuntu 20.04 LTS.  
As said on machine where proxy not needed running about 14 containers in rootless mode, everything works well.

On machines where proxy is needed the proxy container runs  but apparently cannot connect to other containers to route traffic to them.  
nginx logs no errors but also no indication that it gets a connection (there should be log entries for incoming connection).
Letsencrypt logs:
2020-09-24 17:19:19,266:INFO:simp_le:1450: Generating new certificate private key
2020-09-24 17:19:56,272:ERROR:simp_le:1417: CA marked some of the authorizations as invalid, which likely means it could not access http://example.com/.well-known/acme-challenge/X. Did you set correct path in -d example.com:path or --default_root? Are all your domains accessible from the internet?
And when I try to access that domain with wget https://my_domain.com :
Connecting to my_domain.com(my_domain.com)|xxx.xx.xx.xxx|:443... connected.
HTTP request sent, awaiting response... 
and it stays stuck there.

I expose nginx on ports 80 and 443.
Of course all this after having started docker with:
systemctl --user start docker.

Sorry for the long answer, not even sure if your project is the same as the docs I've indicated at the beginning or even linked but looks very similar and I have rootlesskit.sh installed after having followed the instructions in the link given at the beginning of this story of my life.
Again sorry for long answer.
